Use "Unix Makefiles" as the default generator.  Without this patch, the
default generator is "NMake Makefiles", hence requires Microsoft Visual
Studio.  Analogous to Msys2, mingw-w64-cmake,
0005-Default-to-ninja-generator.patch.

diff -Nru cmake-3.22.1-orig/Source/cmake.cxx cmake-3.22.1-patched/Source/cmake.cxx
--- cmake-3.22.1-orig/Source/cmake.cxx	2021-12-07 10:44:21.000000000 -0500
+++ cmake-3.22.1-patched/Source/cmake.cxx	2022-03-23 15:35:25.386828127 -0400
@@ -2174,7 +2174,10 @@
       return gen;
     }
   }
-#if defined(_WIN32) && !defined(__CYGWIN__) && !defined(CMAKE_BOOT_MINGW)
+#if defined(_WIN32) && defined(__MINGW32__) && !defined(CMAKE_BOOT_MINGW)
+  return std::unique_ptr<cmGlobalGenerator>(
+    cm::make_unique<cmGlobalUnixMakefileGenerator3>(this));
+#elif defined(_WIN32) && !defined(__CYGWIN__) && !defined(CMAKE_BOOT_MINGW)
   std::string found;
   // Try to find the newest VS installed on the computer and
   // use that as a default if -G is not specified
